src/qt/walletframe.cpp

```diff
@@ -94,6 +94,10 @@ void WalletFrame::removeWallet(WalletModel* wallet_model)
     WalletView *walletView = mapWalletViews.take(wallet_model);
     walletStack->removeWidget(walletView);
     delete walletView;
+
+    if (mapWalletViews.empty()) {
+        gui->gotoOverviewPage();
```

Comment:
I'd avoid using `gui`, it should be unknown to `WalletFrame` - actually I have a branch to remove `gui` here. May I suggest to do this in `BitcoinGUI::removeWallet`, in `m_wallet_selector->count() == 0` condition.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18896#issuecomment-625035798).